src/Cli/dotnet/Installer/Windows/ElevationContextBase.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.Versioning;
+
+namespace Microsoft.DotNet.Installer.Windows
+{
+    /// <summary>
+    /// Encapsulates information to manage process elevation.
+    /// </summary>
+#if NETCOREAPP
+    [SupportedOSPlatform("windows")]
+#endif
+    internal abstract class ElevationContextBase
+    {
+        /// <summary>
+        /// Gets whether the current process has already been elevated.
+        /// </summary>
+        public bool HasElevated
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// <see langword="true"/> if the the current user has elevated permissions.
+        /// </summary>
+        public abstract bool IsElevated
```

Comment:
HasElevated is used in the ElevationContext to avoid elevating more than once. That way, all the methods that end up on paths requiring privileged operations can just always call Elevate() and leave the work to the ElevationContext.

If we're not elevated, we may need to elevated, but if we are elevated, we won't need to elevate again. The other difference is that IsElevated always queries the security principal, whereas HasElevated is a flag we specifically set in the code. We could have tried to elevate, and the user could have rejected elevation

Reply:
Makes sense, can we include some of that context in the comments?

Reply:
I've pushed a fix with updated comments